zerver/lib/message.py

```diff
@@ -847,18 +851,14 @@ def apply_unread_message_event(user_profile: UserProfile,
     if 'mentioned' in flags:
         state['mentions'].add(message_id)
 
-# 4 hours is probably frequent enough to encourage upgrade
-# and largely mitigates the cache-miss problem. We also
-# don't need to restrict messages to exactly
-# message_visibility_limit. The query is also reasonably fast
-# so cache misses can be managed without cron job.
-@cache_with_key(realm_last_visible_message_id_cache_key, timeout=3600*4)
-def get_first_visible_message_id(realm: Realm) -> int:
-    if realm.message_visibility_limit is None:
-        return 0
+def estimiate_recent_messages(realm: Realm, days: int) -> int:
```

Comment:
Let's have this take a number of hours, not days; that way we can look back a little farther than a day (like 30 hours), so that the lookback periods are sure to overlap, and avoid having to think about edge effects.